## 1º Workshop projeto de interdisciplinaridade FAPERJ

## Pergunta de Pesquisa

**Quais são as características mais comuns a quem dorme por mais horas?**

(tempo de tela, minutos de atividade física, nivel de estresse, alimentação)

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("exemplo_dataset.csv")
df.head()     # df[:5]

,id,horas_sono,tempo_tela,min_atividade_fisica,nivel_estresse,copos_agua,alimentacao,sexo
0,1,7.0,7,4,2.0,2,Regular,F
1,2,6.0,1,80,3.0,3,Boa,M
2,3,8.0,1,62,3.0,3,Boa,M
3,4,6.0,8,34,2.0,4,Regular,M
4,5,5.0,10,51,1.0,6,Regular,M


In [3]:
df.columns

Index(['id', 'horas_sono', 'tempo_tela', 'min_atividade_fisica',
       'nivel_estresse', 'copos_agua', 'alimentacao', 'sexo'],
      dtype='str')

## Classificação das variáveis associadas à pergunta de pesquisa

In [4]:
df["tempo_tela"].unique()

array([ 7,  1,  8, 10,  3,  4,  5,  2, 11,  6,  9])

Tempo de tela é uma variável **numérica** e no dataset varia de 1 a 11. Não há unidade de medida, mas diante dos valores, é possível presumir que sejam horas dado que as pessoas em geral não ficam 1 ou 2 minutos usando telas, mas no mínimo 1 ou 2 horas.

In [5]:
df["min_atividade_fisica"].unique()

array([  4,  80,  62,  34,  51,  26,  19, 102, 118, 117,  55,  85, 109,
        22, 114,  50,  12,  95,  84,  89,  68,  72, 106,  24,  41,  39,
        47,  49,  99,  82,  58,  25,  15,  61,   1,  74, 110,   6,  76,
        97,  98,  60,  16,  46,  28,  18,  83,   2, 108, 113,   9,  52,
        42,  65,   7,  43,  67,  38, 104,  57,  90,  14,  66,   8,   0,
        81,  37,  13, 119, 112,  96,  17, 100,  79,  93,  35, 116,  53,
        48,  71,  21,  10, 107,  73,  59,  33, 105,  63, 103,  92,  20,
        77,  70,  75,  31,  40, 101, 115, 111])

In [6]:
df["min_atividade_fisica"].max()

np.int64(119)

In [7]:
df["min_atividade_fisica"].min()

np.int64(0)

Minutos de atividade física é uma variável **numérica** e varia de 0 a 119 no dataset.

In [8]:
df["nivel_estresse"].unique()

array([ 2.,  3.,  1.,  4.,  5., nan])

Nível de estresse é uma variável **categórica** e varia de 1 a 5 no dataset, logo é possível assumir que 1 seja o nível mínimo de estresse e 5 seja o máximo. Além disso, há também o valor "nan" na tabela, isso quer dizer que há dado faltante na coluna nivel_estresse, a seguir vamos descobrir quantos e decidir se apagamos as linhas ou substituímos pela média dos valores da coluna, por exemplo.

In [9]:
df["alimentacao"].unique()

<StringArray>
['Regular', 'Boa', ' Ruim ', 'boa ']
Length: 4, dtype: str

Alimentação é uma variável **categórica** e possui 4 classes no dataset. No entanto, a categoria "Boa" está escrita de duas formas diferentes, ou seja, as classes "Boa" e "boa" são a mesma coisa. Logo, há, na verdade, apenas 3 categorias para esta variável.

## Verificando a qualidade dos dados

In [10]:
# Verificando se há duplicatas

df.duplicated().sum()

np.int64(0)

In [11]:
# Se houver dados duplicados, verificamos quais são:

df[df.duplicated()]

,id,horas_sono,tempo_tela,min_atividade_fisica,nivel_estresse,copos_agua,alimentacao,sexo


In [12]:
# Verificando se há valores faltantes

dados_faltantes_soma = df.isnull().sum()
dados_faltantes_porc = df.isnull().mean() * 100

df_dados_faltantes = pd.DataFrame({"Dados faltantes (soma)": dados_faltantes_soma, "Dados faltantes (porcentagem)": dados_faltantes_porc})
df_dados_faltantes

,Dados faltantes (soma),Dados faltantes (porcentagem)
id,0,0.0
horas_sono,1,0.5
tempo_tela,0,0.0
min_atividade_fisica,0,0.0
nivel_estresse,1,0.5
copos_agua,0,0.0
alimentacao,0,0.0
sexo,0,0.0
